code/game/gamemodes/dynamic/dynamic.dm

```diff
@@ -793,6 +812,51 @@ GLOBAL_VAR_INIT(dynamic_forced_threat_level, -1)
 	if (!isnull(threat_log))
 		log_threat(-cost, threat_log, reason)
 
+/// Checks if a high impact ruleset was executed and isn't dead. Only one will run at a time in most circumstances.
+/datum/game_mode/dynamic/proc/high_impact_ruleset_active()
+	. = FALSE
+	for(var/datum/dynamic_ruleset/ruleset in executed_rules)
+		if(!CHECK_BITFIELD(ruleset.flags, HIGH_IMPACT_RULESET))
+			continue
+		if(!high_impact_major_event_occured && ruleset.is_dead())
+			continue
+		return TRUE
+
+/// Checks if the station is considered "intact", for the purpose of rolling rulesets with the INTACT_STATION_RULESET flag.
+/datum/game_mode/dynamic/proc/is_station_intact()
+	if(!COOLDOWN_FINISHED(src, intact_cache_expiry))
+		return cached_station_intact
+	COOLDOWN_START(src, intact_cache_expiry, 5 MINUTES)
+	var/old_cached_station_intact = cached_station_intact
+	cached_station_intact = TRUE
+	var/min_pop = CONFIG_GET(number/dynamic_minimum_living_population)
+	if(min_pop)
+		var/total = 0
+		var/living = 0
+		for(var/mob/living/player in GLOB.player_list)
+			if(!is_station_level(player.z))
+				continue
+			if(player.stat < DEAD)
```

Comment:
This just seems way too iffy of a check, since it would include noncrew players, antagonists, etc. I'm not really a fan of the living player check being tacked onto station integrity. We already have antag percent checks which would limit injections when too many crew die, so just integrity should be sufficient.

Reply:
What would be a better way to handle this?

Reply:
imo removing it

Reply:
I ended up removing it and forgetting I did so anyways lol